### DISCUSSION TITLE
Does ruby-kafka supports Protobuf serialization?

### DISCUSSION THREAD

**Discussion by xiaoronglv** [zendesk/ruby-kafka#886] [Q&A] [answered]
https://github.com/zendesk/ruby-kafka/wiki/Encoding-messages-with-Avro

it supports Avro, does it support protobuf?

could you please provide an example about encoring messages with protobuf?

**Comment by geoff2k** [comment, marked as answer]
Note that in the example you linked, ruby-kafka isn't doing any of the avro-related work itself: when producing, the producer is passed the data that has been encoded using AvroTurf, and when consuming, the raw `message.value` is passed back into AvroTurf for decoding.

The same should apply to protobuf -- as long as you can generate a protobuf encoded payload, you  can call ruby-kafka's `#produce` to produce it to a topic, and on the consuming end, you can use protobuf's decoding to turn the `message.value` back into an instance.
